test/readme.md

```diff
@@ -0,0 +1,116 @@
+# Cilium Test Suite
+
+Cilium test suite is based on [Ginkgo test
+framework](https://onsi.github.io/ginkgo/), before start writing test, maybe it
+is a good idea to read [Ginkgo
+Quickstart](https://onsi.github.io/ginkgo/#getting-started-writing-your-first-test)
+guide and run this [test suite
+example](https://github.com/onsi/composition-ginkgo-example) written by the
+[Ginkgo creator](https://github.com/onsi/).
+
+So, before writing test, we should understand the test suite flavors that have
+the system:
+
+- Runtime: This is the flavor where all cilium features are tested.
+- K8S: This is the flavor that we use to test Kubernetes specified features.
+
+Each flavor will power on a new virtual server (vagrant box), ginkgo will
+connect to this server using ssh and run the test specified for that flavor.
+
+To detect what flavor we are going to run, `BeforeSuite` in the test_suite.go
+will identify the scope based on the focus flag string, and it will power on
+the specified server
+
+When the server is power on, the normal execution of the test will start. On
+each spec, a new ssh connection will be open to the virtual server. Mainly this
+connections are wrapped in some helpers, and a Node attribute is exported in
+case that we need to run a specific command on a node.
+
+## Runtime Test:
+
+Runtime test is where all features are tested. It is provisioned based on a
+[ubuntu 17.04 server with all cilium dependencies already in
+place](https://github.com/eloycoto/cilium_basebox).
+
+To run these tests, a --focus flag need to be provided in the Ginkgo command,
+if the focus flag start with  "Run" the system will create runtime vagrant
+box. You can see the status of the box using the following command:
+
+```
+vagrant status runtime
+````
+
+To run the test, you need to execute the following command:
+```
+cd test; ginkgo --focus "Run"
```

Comment:
I think that this requires passwordless "sudo" privileges---might be something worth documenting?